astroid/nodes/node_classes.py

```diff
@@ -4781,33 +4784,59 @@ def get_children(self):
     def _infer(
         self, context: InferenceContext | None = None, **kwargs: Any
     ) -> Generator[InferenceResult, None, InferenceErrorInfo | None]:
-        yield from self._infer_from_values(self.values, context)
+        if self.values:
+            yield from self._infer_with_values(context)
+        else:
+            yield Const("")
+
+    def _infer_with_values(
+        self, context: InferenceContext | None = None, **kwargs: Any
+    ) -> Generator[InferenceResult, None, InferenceErrorInfo | None]:
+        uninferable_already_generated = False
+        for inferred in self._infer_from_values(self.values, context):
+            failed = (
+                inferred is util.Uninferable
+                or isinstance(inferred, Const)
+                and MISSING_VALUE in inferred.value
+            )
+            if failed and self.FAIL_ON_UNINFERABLE:
+                if not uninferable_already_generated:
+                    uninferable_already_generated = True
+                    yield util.Uninferable
+                    continue
+            yield inferred
 
     @classmethod
     def _infer_from_values(
         cls, nodes: list[NodeNG], context: InferenceContext | None = None, **kwargs: Any
     ) -> Generator[InferenceResult, None, InferenceErrorInfo | None]:
         if not nodes:
-            yield
             return
         if len(nodes) == 1:
-            yield from nodes[0]._infer(context, **kwargs)
+            for node in cls._safe_infer_from_node(nodes[0], context, **kwargs):
+                if isinstance(node, Const):
+                    yield node
+                    continue
+                yield Const(MISSING_VALUE)
             return
-        uninferable_already_generated = False
-        for prefix in nodes[0]._infer(context, **kwargs):
+        for prefix in cls._safe_infer_from_node(nodes[0], context, **kwargs):
```

Comment:
Ooh, good spot. If we isolate this change I bet we could get quick consensus to merge it 👍

Reply:
it's isolated now